### ISSUE TITLE
Dask unable to handle pytorch tensors

### ISSUE THREAD

**Issue by estebanag** [dask/dask#9525]
**Problem description**

I am trying to use dask for parallelizing the training of a machine learning model. For this, I make use of pytorch tensors for the model parameters (with [`requires_grad=True`](https://pytorch.org/docs/stable/generated/torch.Tensor.requires_grad.html)). This works fine with the `single-threaded` scheduler. However, when using the `distributed` scheduler, I get:

```
torch/_tensor.py:1083: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at  aten/src/ATen/core/TensorBody.h:482.)
  return self._grad
```
As a consequence, the model never learns.

**Example**

This reproduces the error:

```python
import dask
import torch
from dask.distributed import Client


def func(x):
    return x + 2


def run(single_threaded: bool = True):
    # Set up dask scheduler
    _ = Client()
    if single_threaded:
        dask.config.set(scheduler="single-threaded")

    # Define tensor
    x = torch.linspace(0, 5, 10, dtype=torch.double, requires_grad=True)

    # Run function with dask
    result = []
    for xx in x:
        result.append(dask.delayed(func)(xx))
    _ = dask.compute(*result)


if __name__ == "__main__":
    run(single_threaded=True) 
    print("====================")
    run(single_threaded=False)  # this throws the warning
```

**Environment**

- Dask version: dask==2022.9.1; distributed==2022.9.1; torch==1.12.1
- Python version: 3.8.10
- Operating System: Ubuntu 20.04.3 LTS
- Install method (conda, pip, source): pip

**Comment by estebanag** [comment]
Here's a simple regression example to reproduce the problem (model: `y = slope * x`), where `x` and `y` are given and we wish to estimate `slope`:

```python
import dask
import numpy as np
import torch
import torch.nn as nn
from dask.distributed import Client
from torch.optim import SGD


def func(coef, x):
    return coef * x


def run(single_threaded: bool = True):
    # Simulate x and y
    x = np.linspace(0, 5, 100)
    slope = 3
    y = slope * x

    # Prepare tensors
    coef = torch.zeros(1, dtype=torch.double, requires_grad=True)
    y = torch.tensor(y, requires_grad=False)
    x = torch.tensor(x, requires_grad=False)

    # Set up dask scheduler
    _ = Client()
    if single_threaded:
        dask.config.set(scheduler="single-threaded")

    # Estimate slope
    opt = SGD([coef], lr=0.1)
    for _ in range(300):
        # Predict for current coef (using dask)
        dask_predictions = []
        for xx in x:
            dask_predictions.append(dask.delayed(func)(coef, xx))
        dask_predictions = dask.compute(*dask_predictions)

        predictions = torch.stack(dask_predictions).squeeze()

        # Compute loss
        loss_func = nn.MSELoss()
        loss = loss_func(predictions, y)

        # Backpropagation step
        opt.zero_grad()
        loss.backward()
        opt.step()

    print(f"Slope: {slope}")
    print(f"Estimated slope: {coef[0]}")


if __name__ == "__main__":
    run(single_threaded=True)
    run(single_threaded=False)
```

The output is:
```
Slope: 3
Estimated slope: 2.9999999999999996

Slope: 3
Estimated slope: 0.0
```
The last `Estimated slope: 0.0` corresponds to the estimation of the `slope` when I use the `distributed` scheduler.

@muammar would you know why this is? Considering your experience with [this issue](https://github.com/dask/distributed/issues/2581).
